File: java/logs-logs-logs/src/main/java/LogLevel.java
import java.util.Arrays;
import java.util.Objects;

public enum LogLevel {
    UNKNOWN(null, 0),
    TRACE("TRC", 1),
    DEBUG("DBG", 2),
    INFO("INF", 4),
    WARNING("WRN", 5),
    ERROR("ERR", 6),
    FATAL("FTL", 42);

    private final String levelString;
    private final int encodedLogLevel;

    public String getLevelString() {
        return levelString;
    }

    public int getEncodedLevel() {
        return encodedLogLevel;
    }

    LogLevel(String level, int encodedLevel) {
        levelString = level;
        encodedLogLevel = encodedLevel;
    }

    public static LogLevel fromString(String level) {
        return Arrays.stream(LogLevel.values())
                .filter(logLevel -> Objects.equals(logLevel.getLevelString(), level))
                .findFirst().orElse(UNKNOWN);
    }
}

File: solutions/elixir/log-level/1/lib/log_level.ex
defmodule LogLevel do
  @levels %{0 => :trace, 1 => :debug, 2 => :info, 3 => :warning, 4 => :error, 5 => :fatal}
  def to_label(level, legacy?) do
    cond do
      (level == 0 or level == 5) and legacy? ->
        :unknown

      true ->
        Map.get(@levels, level, :unknown)
    end
  end

  def alert_recipient(level, legacy?) do
    label = to_label(level, legacy?)

    cond do
      label == :error or label == :fatal -> :ops
      label == :unknown and legacy? -> :dev1
      label == :unknown -> :dev2
      true -> false
    end
  end
end
